File: c++/2473.cpp
#include <algorithm>
#include <iostream>

using namespace std;

long long arr[5000];
long long N;
long long minVal = 1e18;
long long a, b, c;

int main() {
  freopen("example.txt", "r", stdin);
  cin >> N;
  for (int i = 0; i < N; i++) {
    cin >> arr[i];
  }

  sort(arr, arr + N);

  for (int start = 0; start < N - 2; start++) {
    int mid = start + 1;
    int end = N - 1;

    while (mid < end) {
      long long sum = arr[start] + arr[mid] + arr[end];
      if (abs(sum) < minVal) {
        minVal = abs(sum);
        a = arr[start];
        b = arr[mid];
        c = arr[end];
      }

      if (sum < 0)
        mid++;
      else
        end--;
    }
  }

  cout << a << ' ' << b << ' ' << c;

  return 0;
}

File: js/2473.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const arr = inputs[1]
  .split(' ')
  .map(Number)
  .sort((a, b) => a - b);

let min = Infinity;
const answer = Array(3).fill(0);

for (let start = 0; start < N - 2; start++) {
  let mid = start + 1;
  let end = N - 1;
  while (mid < end) {
    const sum = arr[start] + arr[mid] + arr[end];
    if (Math.abs(sum) < min) {
      min = Math.abs(sum);
      answer[0] = arr[start];
      answer[1] = arr[mid];
      answer[2] = arr[end];
    }
    if (sum < 0) mid++;
    else end--;
  }
}

console.log(answer.join(' '));
